### DISCUSSION TITLE
Data Store

### DISCUSSION THREAD

**Discussion by gaetanoi** [Infineon/optiga-trust-m#72] [Q&A] [answered]
Hi,

I store application data in arbitrary data objects (type 3 0xF1D0-0xF1DB), I'm not clear if I can modify only few bytes inside a slot without erase the slot (140 bytes). 
What happens if power loss occur while write command is executing? are the old data retained?

regards 
Gaetano

**Comment by ayushev** [comment, marked as answer]
Hi @gaetanoi 

you can define how would you like to write data into the object, either overwrite completely or only partially.
[`optiga_util_write_data()`](https://infineon.github.io/optiga-trust-m/optiga__util_8h.html#a908d07933cfc39b1cfc6c98115baa81c) fnction has a parameter `write_type` which can be either `OPTIGA_UTIL_WRITE_ONLY` or `OPTIGA_UTIL_ERASE_AND_WRITE`, the latter one will erase the object and then write your newly provided data, the first one will simply write over existing data.

**Reply by gaetanoi** [reply]
Hi @ayushev 

Thank you for your reply,

What happens if power loss occur while write command is executing? are the old data retained?

regards
Gaetano

**Reply by ayushev** [reply]
In general data is updated atomically, but there is a nuance:
If it is using SetDataObject ([`optiga_util_write_data()`](https://infineon.github.io/optiga-trust-m/optiga__util_8h.html#a908d07933cfc39b1cfc6c98115baa81c)) -> up to maximum size of the data object, it is atomic
If it is using SetObjectProtected ([`optiga_util_protected_update_start/continue/final`](https://infineon.github.io/optiga-trust-m/optiga__util_8h.html#a936ca9753adb87d7717444a025c87e6d))-> it is **not** atomic. but Version flag reflect unusable state until the completion of protected update
